crates/league-modpkg/src/lib.rs

```diff
@@ -1 +1,72 @@
+use chunk::ModpkgChunk;
+use license::ModpkgLicense;
+use std::collections::HashMap;
 
+mod chunk;
+mod error;
+mod license;
+mod read;
+
+#[derive(Debug, PartialEq)]
+pub struct Modpkg {
+    name: String,
+    display_name: String,
+    description: Option<String>,
+    version: String,
+    distributor: Option<String>,
+    authors: Vec<ModpkgAuthor>,
+    license: ModpkgLicense,
+
+    chunks: HashMap<u64, ModpkgChunk>,
```

Comment:
Some way to authenticate who made mod would be nice like embeded signature.
Maybe just throw in option for pkcs7 blob at the end here?

Reply:
> Would this be the signature that we use to validate the mod when installing it ?

yeah we could validate the mod at install time to ensure authorshio

Reply:
We still need the second signature as a chunk right ?

Reply:
signature verifies chunk data so it can't be a chunk itself it has to be on same level as chunk list
string table could be a chunk (then we get added benefit of compression for it as well)

Reply:
Assuming you opt to keep metadata and strings table stored as chunks:

```
u64 magic
u32 count_of_chunks
u32 size_of_signature // 0 if no signature
[ModpkgChunk chunk_descriptors] // at least allways 2 (metadata and strings table)
[u8 chunk data] // by convention place metadata and strings first so streaming readers can read it early
[u8 signature] // signature always at the end ?
```
Do you see anything wrong with this approach ?